src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1465,6 +1467,117 @@ class Simplex_tree {
     }
   }
 
+  /** \brief Retrieve good values for extended persistence, and separate the 
+   * diagrams into the ordinary, relative, extended+ and extended- subdiagrams. 
+   * Need extend_filtration to be called first!
+   * @param[in] dgm Persistence diagram obtained after calling this->extend_filtration 
+   * and this->get_persistence.
+   * @return A vector of four persistence diagrams. The first one is Ordinary, the 
+   * second one is Relative, the third one is Extended+ and the fourth one is Extended-.
```

Comment:
I don't know how obvious it is what Extended + vs - are. Actually, even for the other categories, a notebook showing the categories with pictures will be nice, after this gets in. Is there some paper we could cite somewhere?

Reply:
I added a link. I have also planned to prepare a notebook illustrating this.